src/selectors/operators/numeric/BinaryNumericOperator.js

```diff
@@ -4,21 +4,24 @@ import Sequence from '../../dataTypes/Sequence';
 import Selector from '../../Selector';
 import createAtomicValue from '../../dataTypes/createAtomicValue';
 
+import YearMonthDuration from '../../dataTypes/valueTypes/YearMonthDuration';
+import DayTimeDuration from '../../dataTypes/valueTypes/DayTimeDuration';
+
 function executeOperator (kind, a, b) {
-    switch (kind) {
-        case '+':
-            return a + b;
-        case '-':
-            return a - b;
-        case '*':
-            return a * b;
-        case 'div':
-            return a / b;
-        case 'idiv':
-            return Math.trunc(a / b);
-        case 'mod':
-            return a % b;
-    }
+	switch (kind) {
```

Comment:
Ah, I did not understand you. Yeah, that is better. To prevent making six different constructors, could you make it like CompareExpression? That one builds a function which executes a compare, including the necessary casts.